Client-side sources not reloaded when switching git branch

**Description**
While developing locally, any source changes trigger a rebuild automatically. But when changing source files by switching git branches, this doesn't happen and the developer is stuck with the sources from the previous branch.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to Helix (Pages) project
1. `git checkout <some-branch>`
2. `hlx up`
3. Make a change to a CSS or JS file
4. Reload browser to see changes
5. Commit & push
5. `git checkout <other-branch>`
6. Reload browser to original state
7. JS/CSS source changes not reflected

**Expected behavior**
Switching between branches should refresh client-side source files in the same way manual changes do.

**Workaround**
- Restart `hlx up`

**Version:**
7.5.2

Not exactly sure if this is related, but more recently I noticed when switching branches on a Helix Pages repo, the dev server does attempt to restart after detecting a change in helix-query.yaml (note there's no actual change in that file required to reproduce it). I assume this would fix the issue of loading the content from the correct branch, but then it fails to come back up:
```
helix-query.yaml modified. Restarting dev server...
[hlx] info: Local Helix Dev server stopped.
[git] info: http: server stopped.
[ERROR] encountered unhandled promise rejection at process level: [object Promise], reason: Error: Invalid configuration:
A list of strains and a strain with the name "default" is required.
    at ConfigValidator.assetValid (/usr/local/lib/node_modules/@adobe/helix-cli/node_modules/@adobe/helix-shared/src/ConfigValidator.js:89:13)
    at HelixConfig.validate (/usr/local/lib/node_modules/@adobe/helix-cli/node_modules/@adobe/helix-shared/src/HelixConfig.js:83:27)
    at HelixConfig.init (/usr/local/lib/node_modules/@adobe/helix-cli/node_modules/@adobe/helix-shared/src/HelixConfig.js:88:16)
```

--- a/src/abstract.cmd.js
+++ b/src/abstract.cmd.js
@@ -90,11 +90,15 @@ class AbstractCommand extends EventEmitter {
     if (!this._initialized) {
       return this.init();
     }
-    this._helixConfig = await (new HelixConfig()
+    this._helixConfig = new HelixConfig()
       .withLogger(this._helixConfig.log)
       .withConfigPath(this._helixConfig.configPath)
-      .withDirectory(this._helixConfig.directory)
-      .init());
+      .withDirectory(this._helixConfig.directory);
+    if (!await this._helixConfig.hasFile()) {
+      // set default config
+      this._helixConfig.withSource(await ConfigUtils.createDefaultConfig(this.directory));
+    }
+    await this._helixConfig.init();
     this._indexConfig = await (new IndexConfig()
       .withLogger(this._indexConfig.log)
       .withConfigPath(this._indexConfig.configPath)

--- a/src/up.cmd.js
+++ b/src/up.cmd.js
@@ -127,7 +127,10 @@ class UpCommand extends BuildCommand {
     let timer = null;
     let modifiedFiles = {};
 
-    this._watcher = chokidar.watch(['src', 'cgi-bin', HELIX_CONFIG, HELIX_QUERY, GIT_HEAD], {
+    this._watcher = chokidar.watch([
+      'src', 'cgi-bin', '.hlx/pages/master/src', '.hlx/pages/master/cgi-bin',
+      HELIX_CONFIG, HELIX_QUERY, GIT_HEAD,
+    ], {
       ignored: /(.*\.swx|.*\.swp|.*~)/,
       persistent: true,
       ignoreInitial: true,

--- a/src/up.cmd.js
+++ b/src/up.cmd.js
@@ -22,6 +22,7 @@ const pkgJson = require('../package.json');
 
 const HELIX_CONFIG = 'helix-config.yaml';
 const HELIX_QUERY = 'helix-query.yaml';
+const GIT_HEAD = '.git/HEAD';
 
 class UpCommand extends BuildCommand {
   constructor(logger) {
@@ -126,7 +127,7 @@ class UpCommand extends BuildCommand {
     let timer = null;
     let modifiedFiles = {};
 
-    this._watcher = chokidar.watch(['src', 'cgi-bin', HELIX_CONFIG, HELIX_QUERY], {
+    this._watcher = chokidar.watch(['src', 'cgi-bin', HELIX_CONFIG, HELIX_QUERY, GIT_HEAD], {
       ignored: /(.*\.swx|.*\.swp|.*~)/,
       persistent: true,
       ignoreInitial: true,
@@ -148,7 +149,7 @@ class UpCommand extends BuildCommand {
     });
   }
 
-  async run() {
+  async setup() {
     await super.init();
     // check for git repository
     if (!await fse.pathExists(path.join(this.directory, '.git'))) {
@@ -235,9 +236,6 @@ class UpCommand extends BuildCommand {
         });
         this.config.strains.get('default').static.url = this.helixPages.staticURL;
       }
-
-      // pretend to have config file to suppress message below
-      hasConfigFile = true;
     } else {
       this.log.info('    __ __    ___         ');
       this.log.info('   / // /__ / (_)_ __    ');
@@ -275,7 +273,10 @@ class UpCommand extends BuildCommand {
     localRepos.forEach((repo) => {
       this._project.registerGitRepository(repo.repoPath, repo.gitUrl);
     });
+  }
 
+  async run() {
+    await this.setup();
     let buildStartTime;
     let buildMessage;
     const onBuildStart = async () => {
@@ -307,7 +308,7 @@ class UpCommand extends BuildCommand {
         if (this._open) {
           opn(`http://localhost:${this._project.server.port}/`, { url: true });
         }
-        if (!hasConfigFile) {
+        if (!await this.config.hasFile() && !await this.helixPages.isPagesProject()) {
           this.log.info(chalk`{green Note:} 
 The project does not have a {cyan helix-config.yaml} which is necessary to 
 access remote content and to deploy helix. Consider running 
@@ -324,13 +325,14 @@ access remote content and to deploy helix. Consider running
 
     this._initSourceWatcher(async (files) => {
       const dirtyConfigFiles = Object.keys(files || {})
-        .filter((f) => f === HELIX_CONFIG || f === HELIX_QUERY);
+        .filter((f) => f === HELIX_CONFIG || f === HELIX_QUERY || f === GIT_HEAD);
       if (dirtyConfigFiles.length) {
         this.log.info(`${dirtyConfigFiles.join(', ')} modified. Restarting dev server...`);
         await this._project.stop();
         await this.reloadConfig();
-        this._project.withHelixConfig(this.config);
-        await this._project.init();
+        await this.setup();
+        // ensure correct module paths
+        this._project.withRuntimeModulePaths([...this.modulePaths, ...module.paths]);
         await this._project.start();
         if (Object.keys(files).length === 1) {
           return Promise.resolve();

